core/src/test/java/org/arquillian/smart/testing/scm/git/GitChangeResolverTest.java

```diff
@@ -59,20 +65,51 @@ public void should_fetch_all_files_from_first_commit_to_given_hash() throws Exce
         this.gitChangeResolver = new GitChangeResolver(gitFolder.getRoot(), "d923b3a", "1ee4abf");
 
         // when
-        final Set<Change> diff = gitChangeResolver.diff();
+        final Set<Change> diff = gitChangeResolver.diff(CUSTOM);
 
         // then
         assertThat(diff).hasSize(18);
     }
 
     @Test
+    public void should_not_applicable_when_git_repository_is_not_initialized() throws IOException {
+        // given
+        gitFolder.delete();
+        gitFolder.create();
```

Comment:
to avoid this workaround, move the tests methods to separated class - eg.: `GitChangeResolverWithoutGitTest` where, in the before method, you will just create the temporary directory and will not unpack the repository.